tecken/apps.py

```diff
@@ -96,32 +97,47 @@ def _redis_store_eviction_policy():
 
     @staticmethod
     def _check_storage_urls():
-        """If you use minio to functionally test S3, since it's
+        """Ensure emulated storage buckets exist.
+
+        If you use minio to functionally test S3, since it's
         ephemeral the buckets you create disappear after a restart.
         Make sure they exist. That's what we expect to happen with the
         real production S3 buckets.
+
+        The emulated GCS may also have missing buckets that can be created
+        during initialization.
         """
         _all_possible_urls = set(
             list(settings.SYMBOL_URLS)
             + [settings.UPLOAD_DEFAULT_URL, settings.UPLOAD_TRY_SYMBOLS_URL]
             + list(settings.UPLOAD_URL_EXCEPTIONS.values())
         )
         for url in _all_possible_urls:
-            if not url or "minio" not in urlparse(url).netloc:
+            if not url:
                 continue
-            bucket = StorageBucket(url)
-            try:
-                bucket.client.head_bucket(Bucket=bucket.name)
-            except ClientError as exception:
-                if exception.response["Error"]["Code"] == "404":
-                    bucket.client.create_bucket(Bucket=bucket.name)
-                    logger.info(f"Created minio bucket {bucket.name!r}")
-                else:
-                    # The most comment problem is that the S3 doesn't match
-                    # the AWS credentials configured.
-                    # If that's the case, this will raise a 403 Forbidden
-                    # ClientError.
-                    raise
+            netloc = urlparse(url).netloc
+            is_minio = "minio" in netloc
+            is_emulated_gcs = "gcs-emulator" in netloc
```

Comment:
"minio" and "gcs-emulator" are magic strings here. I think I would lift them as "constants" with a comment along the lines of "must match docker-compose service names" or something like that.